File: .tmp_py4013/batch_18_dump/3950_exactly_one_consecutive_set_bits_pair.js
// LeetCode 3950 - Exactly One Consecutive Set Bits Pair
// https://leetcode.com/problems/exactly-one-consecutive-set-bits-pair/
var consecutiveSetBits = function(n) {
        let vis = false;
        for (let pre = 0; n > 0; n >>= 1) {
            let cur = n & 1;
            if (pre == cur && cur == 1) {
                if (vis) return false;
                vis = true;
            }
            pre = cur;
        }
        return vis;
    
};

File: .tmp_rs4013/rust_check22/3950_exactly_one_consecutive_set_bits_pair.rs
struct Solution;
// LeetCode 3950 - Exactly One Consecutive Set Bits Pair
// https://leetcode.com/problems/exactly-one-consecutive-set-bits-pair/

impl Solution {
    pub fn consecutive_set_bits(mut n: i32) -> bool {
        let mut vis = false;
        let mut pre = 0;
        while n > 0 {
            let cur = n & 1;
            if pre == cur && cur == 1 {
                if vis {
                    return false;
                }
                vis = true;
            }
            pre = cur;
            n >>= 1;
        }
        vis
    }
}

fn main() {}
